index.js

```diff
@@ -81,7 +81,7 @@ function _parseInput (input, opts, cb) {
 
   var commonPrefix = null
   input.forEach(function (item, i) {
-    if (typeof item === 'string') return
+    if (typeof item === 'string' || Buffer.isBuffer(item) && !item.name) return
```

Comment:
The order of operations is confusion here. Please add parens to make it explicit.